plugin_guide.md

```diff
@@ -0,0 +1,184 @@
+
+# Plugin Guide
+
+The following steps are outlined in this guide:
+
+1. Implementing the Basic Model Interface
+2. Wrapping your BMI in the eWaterCycle interface
+3. Accessing your model through grpc4bmi
+4. Put your model in a container and add it to eWaterCycle 📦
```

Comment:
A few points here: 

1. Would it make sense to combine steps 3 and 4 under "Containerizing your model", as substeps "running gprc4bmi server"& "building a container"? 
2. I would make a separate header for the entrypoint stuff, as that's quite unrelated to building the container.
3. Personally, intuitively, I'd move point 2 (adding the ewatercycle wrapper) until after the container. As building the container has nothing to do with the eWaterCycle wrapper, and the wrapper actually wraps the container (or the local model, but ideally the container).
4. Does it make sense to add a separate header for "(Custom) forcing generation for your model"

Reply:
Overall, I think it would be helpful to make a bit more distinction between the BMI model and the ewatercycle wrapper. Now they are so intertwined that it's hard to see the separate parts and how they play together.